Parsers/ASimRegistryEvent/Parsers/vimRegistryEventMicrosoftWindowsEvent.yaml

```diff
@@ -13,84 +13,141 @@ References:
-                          ActorSessionId = tostring( toint(EventData.SubjectLogonId)), 
-                          ActorDomainName = tostring(EventData.SubjectDomainName),
-                          ActorUserIdType = 'SID',
-                          ActorUsername = tostring(iff (EventData.SubjectDomainName in ('', '-'), EventData.SubjectUserName, strcat(EventData.SubjectDomainName, @"\" , EventData.SubjectUserName))), 
-                          ActorUsernameType = iff(tostring(EventData.ActorDomainName) in ('', '-'),'Simple', 'Windows'),              
-                      // Process 
-                          ActingProcessId = tostring(toint(tostring(EventData.ProcessId))), 
-                          ActingProcessName = tostring(EventData.ProcessName)
-                    | extend     
-                    // -- Aliases
-                          User = ActorUsername,
-                          UserId = ActorUserId,
-                          Dvc = DvcHostname,
-                          Process = ActingProcessName
-                      }; RegistryEvents
\ No newline at end of file
+    let RegistryEvents=(){
+        let ASIM_GetWindowsUserType = (username:string, sid:string) { 
+            case ( 
+                username endswith "$" or sid in ("S-1-5-18", "S-1-5-19", "S-1-5-20"), "Machine"
+                , isempty(username) or sid == "S-1-0-0", ""
+                , "Regular"
+            )
+        };
+        let ASIM_GetAccountType = (sid:string) { 
+            iif ( 
+                sid in ("S-1-0-0","S-1-5-18", "S-1-5-19", "S-1-5-20"), "Simple"
+                , "Windows"
+            )
+        };
+        let ASIM_ParseWindowsEvents = (WindowsEvent:(EventData:dynamic)) {
+            WindowsEvent
+            | extend ActorUsername = strcat(EventData.SubjectDomainName, @'\', EventData.SubjectUserName)
+                , ActorDomainName = tostring(EventData.SubjectDomainName)
+                , ActorUserId = tostring(EventData.SubjectUserSid)
+                , ActorSessionId = tostring(EventData.SubjectLogonId)
+                , ActingProcessName = tostring(EventData.ProcessName)
+                , ActingProcessId = tostring(toint(tolong(EventData.ProcessId)))
+                , RegistryKey = iif(EventData.ObjectName startswith @"\REGISTRY\MACHINE", replace_string(tostring(EventData.ObjectName), @"\REGISTRY\MACHINE","HKEY_LOCAL_MACHINE")
+                , replace_string(tostring(EventData.ObjectName), @"\REGISTRY\USER","HKEY_USERS"))
+                , Type = "WindowsEvent"
+        };
+        let ASIM_ParseSecurityEvents = (SecurityEvent:(SubjectDomainName:string,SubjectUserName:string,ProcessId:string,ObjectName:string,SubjectUserSid:string,SubjectLogonId:string,ProcessName:string)) {
+            SecurityEvent
+            | extend ActorUsername = strcat(SubjectDomainName, @'\', SubjectUserName)
+                , ActorDomainName = SubjectDomainName
```

Comment:
Project rename instead of extend whenever possible.

Reply:
Life can be easy :)

Reply:
Done